src/snowflake/connector/encryption_util.py

```diff
@@ -96,30 +89,17 @@ def encrypt_stream(
             elif len(chunk) % block_size != 0:
                 chunk = PKCS5_PAD(chunk, block_size)
                 padded = True
-            if not use_openssl_only:
-                out.write(data_cipher.encrypt(chunk))
-            else:
-                out.write(encryptor.update(chunk))
+            out.write(encryptor.update(chunk))
         if not padded:
-            if not use_openssl_only:
-                out.write(
-                    data_cipher.encrypt(block_size * chr(block_size).encode(UTF8))
-                )
-            else:
-                out.write(encryptor.update(block_size * chr(block_size).encode(UTF8)))
-        if use_openssl_only:
-            out.write(encryptor.finalize())
+            out.write(encryptor.update(block_size * chr(block_size).encode(UTF8)))
+        out.write(encryptor.finalize())
```

Comment:
the change is copied from the origin PR #1752
but I don't think we need the `finalize()` logic here, I plan to remove it.
for clarity: @sfc-gh-mkeller, @sfc-gh-jdu @sfc-gh-sfan